galsim/table.py

```diff
@@ -280,3 +280,210 @@ def _LookupTable_hash(self):
+
+        >>> print tab2d([0, 1], [2, 3, 4])
+        [[ 2.  3.],
+         [ 3.  4.],
+         [ 4.  5.]]
+
+    Finally, if you want to just evaluate the LookupTable2D at a list of x and y points but without
+    evaluating all possible combinations of x and y values, then you can use the `scatter` keyword
+    with the () operator.
+
+        >>> print tab2d([1, 2], [3, 4], scatter=True)
+        [ 4.  6.]
+
+    The default interpolation method is linear.  Other choices for the interpolant are:
+      - 'floor'
+      - 'ceil'
+
+        >>> tab2d = galsim.LookupTable2D(x, y, z, interpolant='floor')
+        >>> tab2d(2.2, 3.3)
+        5.0
+        >>> tab2d = galsim.LookupTable2D(x, y, z, interpolant='ceil')
+        >>> tab2d(2.2, 3.3)
+        7.0
+
+    The `edge_mode` keyword describes how to handle extrapolation beyond the initial input range.
+    Possibilities include:
+      - 'raise': raise an exception.  (This is the default.)
+      - 'wrap': infinitely wrap the initial range in both directions.
+    In order to use edge_mode='wrap', the first and last column of f, as well as the first and last
+    row of f must match.  (This way we know what the period is in each dimension.)
+
+
+        >>> x = y = np.arange(5)
+        >>> z = x + y[:, np.newaxis]  # function is x + y, dimensions of z is (5, 5)
+        >>> tab2d = galsim.LookupTable2D(x, y, z, edge_mode='raise')
+        >>> tab2d(7, 7)
+        ValueError: Extrapolating beyond input range.
+
+        >>> tab2d = galsim.LookupTable2D(x, y, z, edge_mode='wrap')
+        ValueError: Cannot wrap `f` array with unequal first/last column/row.
+
+    We extend the x and y arrays with a uniform spacing, though any monotonic spacing would work.
+    Note that the [(0,1), (0,1)] argument in np.pad below extends the z array by 0 rows/columns in
+    the leading direction, and 1 row/column in the trailing direction.
+
+        >>> x = np.append(x, x[-1] + (x[-1]-x[-2]))
+        >>> y = np.append(y, y[-1] + (y[-1]-y[-2]))
+        >>> z = np.pad(z, [(0,1), (0,1)], mode='wrap')
+        >>> tab2d = galsim.LookupTable2D(x, y, z, edge_mode='wrap')
+        >>> tab2d(2., 2.)
+        4.0
+        >>> tab2d(2.+5, 2.)  # The period is 5 in either direction.
+        4.0
+        >>> tab2d(2.+15, 2.+35)
+        4.0
+
+    @param x              Strictly increasing array of `x` positions at which to create table.
+    @param y              Strictly increasing array of `y` positions at which to create table.
+    @param f              Ny by Nx input array of function values.
+    @param interpolant    Interpolant to use.  One of 'floor', 'ceil', or 'linear'.
+                          [Default: 'linear']
+    @param edge_mode      Keyword controlling how extrapolation beyond the input range is handled.
+                          See above for details.  [Default: 'raise']
+    """
+    def __init__(self, xs, ys, f=None, interpolant='linear', edge_mode=None):
```

Comment:
Why `f=None` rather than just `f`?  It seems like someone could then do ```LookupTable2D(x,y)``` and get an uninformative error message when the code gets to ```self.f=np.ascontiguousarray(f, dtype=float)```?

Also, why is there `x` vs. `xs`, `y` vs. `ys`?

Should we do a check for strictly increasing x and y?